server/scripts/events/notifyProductListingCreated.js

```diff
@@ -0,0 +1,52 @@
+const { generateScript, integrationSdkInit } = require('../../api-util/scriptManager');
+const { StorageManagerClient } = require('../../api-util/storageManagerHelper');
+const { httpFileUrlToStream } = require('../../api-util/httpHelpers');
+
+const SCRIPT_NAME = 'notifyProductListingCreated';
+const EVENT_TYPES = 'listing/created';
+const RESOURCE_TYPE = 'listing';
+
+const processEvent = async (integrationSdk, event, storageManagerClient) => {
+  const { resourceType, eventType, resourceId, resource } = event.attributes;
+  if (resourceType !== RESOURCE_TYPE || eventType !== EVENT_TYPES) return;
+
+  const { attributes: listing, relationships } = resource;
+  const userId = relationships?.author?.data?.id?.uuid;
+  const listingId = resourceId?.uuid;
+  const imageUrl = listing?.privateData?.transloaditSslUrl;
+
+  if (!imageUrl || !userId || !listingId) return;
+
+  try {
+    const originalAssetData = await storageManagerClient.uploadOriginalAsset(
+      userId,
+      listingId,
+      imageUrl
+    );
+    const imageStream = await httpFileUrlToStream(imageUrl);
+    const { data: sdkImage } = await integrationSdk.images.upload({ image: imageStream });
+
+    await integrationSdk.listings.update(
```

Comment:
After this we need to communicate with Slack to start the approval flow. I'll be taking care of this, but just wanted to point it out